Update README.md to provide notice of deprecation of legacy C++ API. (#163)
uxlfoundation/oneCCL: README.md

- oneAPI, oneCCL and OFI: Path to Heterogeneous Architecure Programming with Scalable Collective Communications: [recording](https://www.youtube.com/watch?v=ksiZ90EtP98&feature=youtu.be) and [slides](https://www.openfabrics.org/wp-content/uploads/2020-workshop-presentations/502.-OFA-Virtual-Workshop-2020-oneCCL-v5.pdf)

## Notice of Deprecation 

> ### Deprecation Of C++ API

Reviewer: ```suggestion
### Deprecation of C++ API
```

> ### Notice of Deprecation Of Existing C++ API

Reviewer 2: simplifying the title to "Deprecation of C++ API" or "Upcoming Deprecation of C++ API" ?

Author: Agreed.  Done.

> In oneCCL version 2021.17 included with the 2025.3 oneAPI release, oneCCL will add support for a new C API that closely follows the NCCL* API standard. The existing C++ API will remain available, and will remain the default API for this release. Details explaining how an application may link against and use the new API will be shared in this release.

Reviewer: ```suggestion
- In oneCCL version 2021.17 included with the 2025.3 oneAPI release, oneCCL will add support for a new C API that closely follows the NVIDIA Collective Communications Libary (NCCL)* API standard. The existing C++ API will remain available and will remain the default API for the 2021.17 release. Details explaining how an application may link against and use the new API will be shared in this release.
```

> Please visit this [RFC](https://github.com/uxlfoundation/oneCCL/tree/rfcs/rfcs/20240806-c-api) to view the proposed API and provide any feedback. 

Reviewer: ```suggestion
See the [oneCCL C API RFC document](https://github.com/uxlfoundation/oneCCL/tree/rfcs/rfcs/20240806-c-api) to view the proposed API and provide any feedback. 
```

> In oneCCL version 2022.0 included with 2026.0 oneAPI release, oneCCL will use the new NCCL like C API by default. This is a breaking change. The legacy C++ API will remain available and details explaining how an application may link against the legacy API are forthcoming.

Author: I checked and NCCL is trademarked.  I will make corrections that reference the trademark.

> In oneCCL version 2022.0 included with 2026.0 oneAPI release, oneCCL will use the new NCCL like C API by default. This is a breaking change. The legacy C++ API will remain available and details explaining how an application may link against the legacy API are forthcoming.

Author: Sounds good.  Done.

> In oneCCL version 2022.0 included with 2026.0 oneAPI release, oneCCL will use the new NCCL like C API by default. This is a breaking change. The legacy C++ API will remain available and details explaining how an application may link against and use the legacy API will be included in this release.

Reviewer: ```suggestion
- In oneCCL version 2022.0 included with the 2026.0 oneAPI release, oneCCL will use the new NCCL like C API by default. This is a breaking change. The legacy C++ API will remain available, and details explaining how an application may link against and use the legacy API will be included in the 2022.0 release.
```

> - In oneCCL version 2022.0 included with the 2026.0 oneAPI release, oneCCL will use the new NCCL like C API by default. This is a breaking change. The legacy C++ API will remain available, and details explaining how an application may link against and use the legacy API will be included in the 2022.0 release.

Reviewer 3: no asterisk here in NCCL

Author: The documentation for the trademark and branding rules explained that the asterisk is required only on the first use of the trademark.

> Please note that application use of both APIs at the same time is not supported.

Reviewer 2: > Please note that application use of both APIs at the same time is not supported.

Note: Applications cannot use both the C and C++ APIs simultaneously.

## Security Policy <!-- omit in toc -->

See [SECURITY](SECURITY.md) for more information.

## Notices and Disclaimers

Intel technologies may require enabled hardware, software or service activation.

No product or component can be absolutely secure.

Your costs and results may vary.

> © Intel Corporation. Intel, the Intel logo, and other Intel marks are trademarks of Intel Corporation or its subsidiaries. Other names and brands may be claimed as the property of others.

Reviewer 3: this covers the trademark issue, is that right ?

Author: Yes.
